asmcli/commands/create-mesh.sh

```diff
@@ -70,105 +76,76 @@ create-mesh_validate_args() {
+    info "Fetching/writing GCP credentials to ${KCF}..."
+    KUBECONFIG="${KCF}" retry 2 gcloud container clusters get-credentials "${CLUSTER_NAME}" \
       --project="${PROJECT_ID}" \
-      --format='value(selfLink)')"
-      context_append "clusterRegistrations" "${CTX_CLUSTER} ${GKE_CLUSTER_URI}"
-    else
-      exit_if_cluster_registered_to_another_fleet "${PROJECT_ID}" "${CLUSTER_LOCATION}" "${CLUSTER_NAME}"
-      warn "Cluster ${CLUSTER_NAME} is already registered with fleet ${FLEET_ID}. Skipping."
-    fi
-    context_append "clusterContexts" "${CTX_CLUSTER}"
+      --zone="${CLUSTER_LOCATION}"
+    
+    # save the kubeconfig to context
+    context_append "kubeconfigFiles" "${KCF}"
   done <<EOF
 $(context_list "clustersInfo")
 EOF
-}
-
-create_mesh() {
-  local CTX_CLUSTER GKE_CLUSTER_URI
 
-  # for-loop does not read lines but words, so setting IFS to explicitly split with line breaks
-  # Also context_list might return an empty list so we use for-loop to bypass that scenario
-  while read -r CTX_CLUSTER GKE_CLUSTER_URI; do
-    add_one_to_mesh "${CTX_CLUSTER}" "${GKE_CLUSTER_URI}"
-  done < <(context_list "clusterRegistrations")
+  # validate clusters are valid
+  while read -r KCF; do
+    context_set-option "KUBECONFIG" "${KCF}"
+    context_set-option "CONTEXT" "$(kubectl config current-context)"
+    is_cluster_registered
+  done <<EOF
+$(context_list "kubeconfigFiles")
+EOF
 }
 
-add_one_to_mesh() {
-  local CTX_CLUSTER; CTX_CLUSTER="${1}"
-  local GKE_CLUSTER_URI; GKE_CLUSTER_URI="${2}"
-  local PROJECT_ID CLUSTER_LOCATION CLUSTER_NAME MEMBERSHIP_NAME
-  IFS='_' read -r _ PROJECT_ID CLUSTER_LOCATION CLUSTER_NAME < <(echo "$CTX_CLUSTER")
-  local FLEET_ID; FLEET_ID="$(context_get-option "FLEET_ID")"
-
-  MEMBERSHIP_NAME="$(generate_membership_name "${PROJECT_ID}" "${CLUSTER_LOCATION}" "${CLUSTER_NAME}")"
-
-  info "Registering the cluster ${PROJECT_ID}/${CLUSTER_LOCATION}/${CLUSTER_NAME} as ${MEMBERSHIP_NAME}..."
-
-  retry 2 gcloud container hub memberships register "${MEMBERSHIP_NAME}" \
-    --project="${FLEET_ID}" \
-    --gke-uri="${GKE_CLUSTER_URI}" \
-    --enable-workload-identity
+create-mesh_register() {
+  while read -r KCF; do
+    context_set-option "KUBECONFIG" "${KCF}"
+    context_set-option "CONTEXT" "$(kubectl config current-context)"
+    is_cluster_registered
+    if ! is_membership_crd_installed; then
+      register_cluster
+    fi
+  done <<EOF
+$(context_list "kubeconfigFiles")
+EOF
 }
 
 install_all_remote_secrets() {
-  local CTX_CLUSTER1 CTX_CLUSTER2
-
-  while read -r CTX_CLUSTER1; do
-    while read -r CTX_CLUSTER2; do
-      if [[ "${CTX_CLUSTER1}" != "${CTX_CLUSTER2}" ]]; then
-        install_one_remote_secret "${CTX_CLUSTER1}" "${CTX_CLUSTER2}"
+  while read -r KCF1; do
+    while read -r KCF2; do
+      if [[ "${KCF1}" != "${KCF2}" ]]; then
+        install_one_remote_secret "${KCF1}" "${KCF2}"
       fi
     done <<EOF
-$(context_list "clusterContexts")
+$(m1context_list "kubeconfigFiles")
```

Comment:
What's the `m1` here?

Reply:
sorry that's remnant from my local tests..